### ISSUE TITLE
SVG transform-origin ignored

### ISSUE THREAD

**Issue by febdoctor** [Kozea/WeasyPrint#2256]
The attribute "transform-origin" sets the reference point used in transformations set by the "transform" attribute. In the examples below, a rotation was used with the origin set to the center of the triangle. In WeasyPrint, the origin is ignored and the triangle is rotated relative to (0, 0). When the object is far from (0, 0), it can rotate far from it's intended location and far outside the viewport.

The following SVG reproduces the problem. The rotation is small in order to make the error more visible.

```
<svg xmlns="http://www.w3.org/2000/svg"  viewBox="0 0 400 400">

<defs>
  <g id="triangle" transform="scale(0.6)">
    <path d="M 0,-20 l 23,40 l -46,0 Z" stroke="purple" fill="none" />
  </g>
</defs>


<use href="#triangle" x="10mm" y="10mm"/>
<use href="#triangle" x="10mm" y="10mm" transform-origin="10mm 10mm" transform="rotate(9)" />

</svg>
```

In Firefox, the triangle is rotated relative to the center of the triangle:

![svgtransformorigin-firefox](https://github.com/user-attachments/assets/0c293bf4-d3a6-4575-8525-90b626f3913d)

But WeasyPrint rotates it relative to (0, 0). It this example it looks slightly offset because the rotation is small.

![svgtransformorigin-weasyprint](https://github.com/user-attachments/assets/7da52c52-6c81-4851-8693-1d0122eecf6e)

Reference files:
[svgtransformorigin-firefox.pdf](https://github.com/user-attachments/files/17058980/svgtransformorigin-firefox.pdf)
[svgtransformorigin.pdf](https://github.com/user-attachments/files/17058981/svgtransformorigin.pdf)

**Comment by liZe** [comment]
Thanks for the bug report.

It is supported by browsers, but it’s not in the SVG specification, even in SVG2. It’s in [this draft](https://www.w3.org/TR/2009/WD-SVG-Transforms-20090320/), overridden by [this CSS module](https://www.w3.org/TR/css-transforms-1/).

**Comment by liZe** [comment]
If anyone is interested in adding this feature, we need to update this function (and the `utils.transform` function it calls) to take care of `transform-origin`:

https://github.com/Kozea/WeasyPrint/blob/a5d8412d0f11e815db31652fdd88fed2fecfeb66/weasyprint/svg/__init__.py#L748-L755

**Comment by siaslas98** [comment]
Hi, is this still available for me to work on?

**Comment by liZe** [comment]
> Hi, is this still available for me to work on?

Yes, of course! Don’t hesitate to ask if you have any questions.
